src/main/kotlin/com/atlassian/performance/tools/aws/Ec2Instance.kt

```diff
@@ -21,4 +21,6 @@ internal class Ec2Instance(
     override fun release(): CompletableFuture<*> {
         return terminationBatchingEc2.terminate(instance.instanceId)
     }
+
+    override fun toString() = "Ec2Instance(instanceId = ${instance.instanceId})"
```

Comment:
Region and status would also help, but this is a huge step forward in diagnosability.